src/compiler/checker.ts

```diff
@@ -36952,7 +36955,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         const isAsync = (getFunctionFlags(func) & FunctionFlags.Async) !== 0;
         forEachYieldExpression(func.body as Block, yieldExpression => {
             const yieldExpressionType = yieldExpression.expression ? checkExpression(yieldExpression.expression, checkMode) : undefinedWideningType;
-            pushIfUnique(yieldTypes, getYieldedTypeOfYieldExpression(yieldExpression, yieldExpressionType, anyType, isAsync));
+            pushIfUnique(yieldTypes, getYieldedTypeOfYieldExpression(yieldExpression, yieldExpression.expression && isConstContext(yieldExpression.expression) ? getRegularTypeOfLiteralType(yieldExpressionType) : yieldExpressionType, anyType, isAsync));

```

Comment:
This line is too long and we haven't turned on a line length yet. I'd suggest extracting the conditional in a similar way as the first diff in the PR:
```suggestion
            let yieldExpressionType = yieldExpression.expression ? checkExpression(yieldExpression.expression, checkMode) : undefinedWideningType;
            if (yieldExpression.expression && isConstContext(yieldExpression.expression)) {
                yieldExpressionType = getRegularTypeOfLiteralType(yieldExpressionType);
            }
            pushIfUnique(yieldTypes, getYieldedTypeOfYieldExpression(yieldExpression, yieldExpression.expression && isConstContext(yieldExpression.expression) ? getRegularTypeOfLiteralType(yieldExpressionType) : yieldExpressionType, anyType, isAsync));
```